ports/rp2/rp2_psram.c

```diff
@@ -0,0 +1,182 @@
+#include "hardware/structs/ioqspi.h"
+#include "hardware/structs/qmi.h"
+#include "hardware/structs/xip_ctrl.h"
+#include "hardware/clocks.h"
+#include "hardware/sync.h"
+#include "rp2_psram.h"
+
+
+void __no_inline_not_in_flash_func(psram_set_qmi_timing)() {
```

Comment:
Now that I wasn't aware of! If we're eventually going to have an `rp2.PSRAM()` module (and I think we should) then timing tweaks and performance tomfoolery settings could go there.

Thank you!

Reply:
It strikes me that for our purposes we want PSRAM to be init *always*, since it replaces SRAM as the MicroPython heap.

Since we can't run MicroPython before initialising the heap, it's not possible for an `rp2.PSRAM()` module invoked by `boot.py` to work for Presto.

Also I have no idea if we can *add* PSRAM to the heap at runtime.

Reply:
> Also I have no idea if we can _add_ PSRAM to the heap at runtime

Yes, that's now possible using MICROPY_GC_SPLIT_HEAP and MICROPY_GC_SPLIT_HEAP_AUTO (the esp32 port does this).